src/osuTK/Platform/X11/XI2MouseKeyboard.cs

```diff
@@ -193,11 +192,15 @@ KeyboardState IKeyboardDriver2.GetState(int index)
             }
         }
 
-        public KeyboardState[] GetStates()
+        public void GetStates(List<KeyboardState> result)
         {
             lock (Sync)
             {
-                return keyboards.Select(device => device.State).ToArray();
+                result.Clear();
+                for (int i = 0; i < keyboards.Count; i++)
+                {
+                    result.Add(((IKeyboardDriver2)this).GetState(i));
```

Comment:
Are these explicit typecasts required in all of these? It was already returning `KeyboardState`s before, which should add into the list just fine, no?

Reply:
Casts are required for disambiguate between `IKeyboardDriver2.GetState` and `IMouseDriver2.GetState` because this class implements both interfaces.

Reply:
Casts are required for disambiguate between IKeyboardDriver2.GetState and IMouseDriver2.GetState because this class implements both interfaces.